Classes/EventStore/Normalizer/ValueObjectNormalizer.php

```diff
@@ -52,6 +54,25 @@ public function supportsDenormalization($data, $className, $format = null): bool
         }
     }
 
+    public function normalize($object, string $format = null, array $context = [])
+    {
+        if (!$object instanceof \JsonSerializable && (!$object instanceof \Stringable || !method_exists($object, '__toString'))) {
+            throw new InvalidArgumentException(sprintf('The object must implement "%s" or "%s".', \JsonSerializable::class, \Stringable::class));
+        }
+
+        if ($object instanceof \JsonSerializable) {
+            return $this->serializer->normalize($object->jsonSerialize(), $format, $context);
+        }
+        if ($object instanceof \Stringable || method_exists($object, '__toString')) {
+            return $this->serializer->normalize((string) $object, $format, $context);
+        }
+    }
+
+    public function supportsNormalization($data, string $format = null)
+    {
+        return ($data instanceof \JsonSerializable || $data instanceof \Stringable || method_exists($data, '__toString'));
```

Comment:
See above.
Also I think we might need an additional `is_object()` check here sind $data could be of any type

Reply:
What case should the is_object cover?

Reply:
Not sure if `method_exists` fails for things like simple types or arrays!?

Reply:
..it would certainly produce wrong results for a string that represents a class name implementing __toString (even if that case is pretty unlikely)